### DISCUSSION TITLE
Understanding theme extension

### DISCUSSION THREAD

**Discussion by DiabolusGX** [squidfunk/mkdocs-material#3929] [General]
### Problem statement:
- Override `tabs`, to hide some tabs from top nav bar (like "Getting Started").
- Override `nav` on a particular file (let's say "Overview") to show hidden tabs ("Getting Started") from the top nav.

I'm struggling with:
1. Identifier on the file such that I can skip that in `tabs` overrides partial and show the same file `nav` override.
2. Override `nav` only if `Overview` tab is active.

**Comment by DiabolusGX** [comment]
@squidfunk 
As discussed on [Gitter](https://gitter.im/squidfunk/mkdocs-material), you suggested to [override the site_nav block](https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks) and then [set this template only for the page](https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-template)
- Should I use `nav` override partial or `site_nav` blocks for this?
- As per my understanding, using `template` meta data will use new template file instead of main.html but I just want to override `nav`, is there any alternative, this approach sounds much more demanding?

**Reply by squidfunk** [reply]
> Should I use nav override partial or site_nav blocks for this?

Both are possible, it depends on where you need to inject your changes. Always try to use the block first, then partial.

> As per my understanding, using template meta data will use new template file instead of main.html but I just want to override nav, is there any alternative, this approach sounds much more demanding?

This is how MkDocs theme extension works, so if you want to change the theme, this is the simplest and most economical way.

**Comment by DiabolusGX** [comment]
Hey, @squidfunk just wanted to know:
1. if I can find reference for `site_nav` block override somewhere? I'm not able to implement cleanly, do I need to write my own css?
2. is it possible to override `site_nav` on a specific tab? like only on Home/Index tab from top nav and keep default site_nav on About tab.

**Reply by squidfunk** [reply]
1. No, just follow the normal process when [overriding blocks](https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks). `site_nav` is defined [here](https://github.com/squidfunk/mkdocs-material/blob/3c174538ab5ae1632490bb2e89796e12e4e1e1ec/src/base.html#L252-L291).
2. Create an additional template, which you call something other than `main.html`, and set it for each page as a [template](https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-template).
